vscode-ext/src/debugger.ts

```diff
@@ -0,0 +1,126 @@
+import * as vscode from 'vscode';
+import * as path from 'node:path';
+import { globalContext } from './extension';
+import { configFilePath, isTargetInFile } from './config';
+import { LAST_TARGET_KEY, TARGETLESS_TARGET_NAME } from './util';
+import { MirrordAPI } from './api';
+import { checkVersion, versionCheckInterval } from './versionCheck';
+
+
+const CI_BUILD_PLUGIN = process.env.CI_BUILD_PLUGIN === 'true';
+
+
+
+/// Get the name of the field that holds the exectuable in a debug configuration of the given type.
+function getExecutableFieldName(config: vscode.DebugConfiguration): keyof vscode.DebugConfiguration {
+	switch (config.type) {
+		case "pwa-node":
+		case "node": {
+			return "runtimeExecutable";
+		}
+		case "python": {
+			if ("python" in config) {
+				return "python";
+			}
+			// Official documentation states the relevant field name is "python" (https://code.visualstudio.com/docs/python/debugging#_python), 
+			// but when debugging we see the field is called "pythonPath".
+			return "pythonPath";
+		}
+		default: {
+			return "program";
+		}
+	}
+
+}
+
+
+export class ConfigurationProvider implements vscode.DebugConfigurationProvider {
+	async resolveDebugConfiguration(folder: vscode.WorkspaceFolder | undefined, config: vscode.DebugConfiguration, token: vscode.CancellationToken): Promise<vscode.DebugConfiguration | null | undefined> {
+
+		if (!globalContext.workspaceState.get('enabled')) {
+			return config;
+		}
+
+		// For some reason resolveDebugConfiguration runs twice for Node projects. __parentId is populated.
+		if (config.__parentId || config.env?.["__MIRRORD_EXT_INJECTED"] === 'true') {
+			return config;
+		}
+
+		// do not send telemetry for CI runs
+		if (vscode.env.isTelemetryEnabled && !CI_BUILD_PLUGIN) {
```

Comment:
The checkVersion logic is split between this file and versionCheck.ts. Can this entire `if` be moved into versionCheck.ts?